File: src/test/java/com/soapdogg/lispInterpreter/printer/AtomNodePrinterTest.java
package com.soapdogg.lispInterpreter.printer;

import com.soapdogg.lispInterpreter.datamodels.AtomNode;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

class AtomNodePrinterTest {

    private AtomNodePrinter atomNodePrinter;

    @BeforeEach
    void setup() {
        atomNodePrinter = new AtomNodePrinter();
    }

    @Test
    void atomNodePrinterTest() {
        String value = "value";
        AtomNode atomNode = Mockito.mock(AtomNode.class);
        Mockito.when(atomNode.getValue()).thenReturn(value);

        String actual = atomNodePrinter.printAtomNode(atomNode);

        Assertions.assertEquals(value, actual);
    }
}

File: src/test/java/com/soapdogg/lispInterpreter/printer/AtomNodePrinterTest.kt
package com.soapdogg.lispInterpreter.printer

import com.soapdogg.lispInterpreter.datamodels.AtomNode
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

class AtomNodePrinterTest {

    private var atomNodePrinter: AtomNodePrinter = AtomNodePrinter()

    @Test
    fun atomNodePrinterTest() {
        val value = "value"
        val atomNode = Mockito.mock(AtomNode::class.java)
        Mockito.`when`(atomNode.value).thenReturn(value)
        val actual = atomNodePrinter.printAtomNode(atomNode)
        Assertions.assertEquals(value, actual)
    }
}